keps/sig-scheduling/20190926-default-even-pod-spreading.md

```diff
@@ -0,0 +1,180 @@
+---
+title: Default Even Pod Spreading
+authors:
+  - "@alculquicondor"
+owning-sig: sig-scheduling
+reviewers:
+  - "@ahg-g"
+  - "@Huang-Wei"
+approvers:
+  - "@ahg-g"
+  - "@k82cn"
+creation-date: 2019-09-26
+last-updated: 2010-09-26
+status: provisional
+see-also:
+  - "/keps/sig-aaa/20190221-even-pods-spreading.md"
+---
+
+# Default Even Pod Spreading
+
+## Table of Contents
+
+<!-- toc -->
+- [Release Signoff Checklist](#release-signoff-checklist)
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+  - [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [API](#api)
+  - [Implementation Details](#implementation-details)
+    - [In the metadata/predicates/priorities flow](#in-the-metadatapredicatespriorities-flow)
+    - [In the scheduler framework](#in-the-scheduler-framework)
+  - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Alternatives [optional]](#alternatives-optional)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+- [ ] kubernetes/enhancements issue in release milestone, which links to KEP (this should be a link to the KEP location in kubernetes/enhancements, not the initial KEP PR)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+## Summary
+
+With `EvenPodsSpreading`, workload authors can define spreading rules for their loads based on the
```

Comment:
Can we use runtime class to implement this feature, we could add another field in the `Scheduling` struct. https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/runtime-class-scheduling.md

Reply:
`RuntimeClass` the objective to deal with different runtimes in a node. This feature doesn't have to do anything with that.